packages/nextjs/components/NetworksDropdown.tsx

```diff
@@ -80,36 +107,97 @@ export const NetworksDropdown = ({ onChange }: { onChange: (options: any) => any
+
   if (!mounted) return null;
   return (
-    <Select
-      defaultValue={groupedOptions["mainnet"].options[0]}
-      instanceId="network-select"
-      options={Object.values(groupedOptions)}
-      onChange={onChange}
-      components={{ Option: IconOption }}
-      isSearchable={!isMobile}
-      className="max-w-xs relative text-sm w-44"
-      theme={theme => ({
-        ...theme,
-        colors: {
-          ...theme.colors,
-          primary25: isDarkMode ? "#401574" : "#efeaff",
-          primary50: isDarkMode ? "#551d98" : "#c1aeff",
-          primary: isDarkMode ? "#BA8DE8" : "#551d98",
-          neutral0: isDarkMode ? "#130C25" : theme.colors.neutral0,
-          neutral80: isDarkMode ? "#ffffff" : theme.colors.neutral80,
-        },
-      })}
-      styles={{
-        menuList: provided => ({ ...provided, maxHeight: 280, overflow: "auto" }),
-        control: provided => ({ ...provided, borderRadius: 12 }),
-        indicatorSeparator: provided => ({ ...provided, display: "none" }),
-        menu: provided => ({
-          ...provided,
-          border: `1px solid ${isDarkMode ? "#555555" : "#a3a3a3"}`,
-        }),
-      }}
-    />
+    <>
+      <Select
+        value={selectedOption}
+        defaultValue={groupedOptions["mainnet"].options[0]}
+        instanceId="network-select"
+        options={Object.values(groupedOptions)}
+        onChange={handleSelectChange}
+        components={{ Option: IconOption }}
+        isSearchable={!isMobile}
+        className="max-w-xs relative text-sm w-44"
+        theme={theme => ({
+          ...theme,
+          colors: {
+            ...theme.colors,
+            primary25: isDarkMode ? "#401574" : "#efeaff",
+            primary50: isDarkMode ? "#551d98" : "#c1aeff",
+            primary: isDarkMode ? "#BA8DE8" : "#551d98",
+            neutral0: isDarkMode ? "#130C25" : theme.colors.neutral0,
+            neutral80: isDarkMode ? "#ffffff" : theme.colors.neutral80,
+          },
+        })}
+        styles={{
+          menuList: provided => ({ ...provided, maxHeight: 280, overflow: "auto" }),
+          control: provided => ({ ...provided, borderRadius: 12 }),
+          indicatorSeparator: provided => ({ ...provided, display: "none" }),
+          menu: provided => ({
+            ...provided,
+            border: `1px solid ${isDarkMode ? "#555555" : "#a3a3a3"}`,
+          }),
+        }}
+      />
+      <dialog id="see-all-modal" className="modal">
+        <div className="flex flex-col modal-box justify-center px-12 h-3/4 sm:w-1/2 max-w-5xl bg-base-200">
+          <div className="flex justify-between items-center mb-6">
+            <h3 className="font-bold text-xl">All Chains</h3>
+            <div className="modal-action mt-0">
+              <button
+                className="btn btn-error"
+                onClick={() => (document.getElementById("see-all-modal") as HTMLDialogElement)?.close()}
+              >
+                Close
+              </button>
```

Comment:
nitpck: lol the btn-error seems stand out a bit : 

![image](https://github.com/BuidlGuidl/abi.ninja/assets/80153681/b5c5f4fe-0e59-4ebf-9680-2464017d21e0)

Maybe we try out some color from theme like primary secondary etc ? Also maybe instead of big close button we could have just "X" a bit subtle and minimal ?